### ISSUE TITLE
CSS Scaffold and PHP7

### ISSUE THREAD

**Issue by eugene-wh** [SocialEngine/phpv4-issues#472]
It looks like a lot of errors/warnings with php7 are caused by CSS Scafford. The last update for it was ~3 years ago.

It might make sense to switch to a something newer, instead of patching an outdated library.

**Comment by Khushboo016** [comment]
Hi @eugene-wh , As you can see from our blog page http://blog.socialengine.com/2016/05/26/socialengine-php-4-8-11-is-released that we've just resolved a major fatal error with PHP 7 and still have to overcome many other issues related to compatibility of SocialEngine with PHP 7. So, it would be great if you can help us in doing that by providing us the details of the issues you are facing while running SocialEngine with PHP 7.

Thanks!

**Comment by eugene-wh** [comment]
> we've just resolved a major fatal error with PHP 7 and still have to overcome many other issues related to compatibility of SocialEngine with PHP 7

Yes, I follow all updates. That was a good job.

>  if you can help us in doing that by providing us the details of the issues you are facing while running SocialEngine with PHP 7.

Are you able to get SE running with php7? When we have run tests SE was not working with php7 at all. Errors I've seen were related to css Scaffold. 
We also have to work with different theme customizations. Css scaffold does not support a lot of modern css things. For example, if you try to use 
```
@keyframes mymove {
   from {top: 0px;}
   to {top: 200px;}
}
```
there will be errors.

I do believe that instead of fixing this library it's more efficient to change it for something more modern and active. Why not switch to less/sass?

**Comment by Elshara** [comment]
Or just use plain css 3.

**Comment by RaymondBenc** [comment]
We are currently running QA on PHP 7 and have ran into this problem. We've applied the suggested [PR](https://github.com/balupton/CSScaffold/pull/3) and it resolved the issues. We are looking into other options as well.

**Comment by eugene-wh** [comment]
@RaymondBenc were you testing SocialEngine or just the Scaffold?

**Comment by RaymondBenc** [comment]
@eugene-wh We are currently testing SocialEngine. Like you mentioned lots of the issues were related to Scaffold and we needed to apply a fix there in order to continue testing SocialEngine.

**Comment by Elshara** [comment]
It's due to the template engine not recognizing the php version I
expect. Updating this might help.

**Comment by hasithaaa** [comment]
I am try to run SocialEngine 4.2.3 on PHP 7.1 . 
PHP Warning:  session_write_close(): Failed to write session data using user defined save handler. (session.save_path: session) . How should  remove this warning.

**Comment by shivani2910** [comment]
Hi @hasithaaa,

SocialEngine PHP is currently not compatible with PHP 7 or 7.1. But we're glad to share that we've already done this work and added compatibility for php v7 and 7.1 in our upcoming upgrade i.e. Se PHP 4.9 which is going to be released very soon.
Stay tuned to our [blog](http://blog.socialengine.com/) for any further information on this.

Thanks,
Shivani

**Comment by RaymondBenc** [comment]
@hasithaaa like @shivani2910 mentioned, support for PHP 7 is coming in 4.9, however could you provide the full stack trace warning on here? I want to confirm to make sure this does not happen on 4.9.

Thanks

**Comment by hasithaaa** [comment]
Thanks for your reply.
Please find full stack trace bellow.


017-03-23T03:59:58+00:00 WARN (4): [2] session_write_close(): Failed to write session data using user defined save handler. (session.save_path: session) (/var/www/html/application/libraries/Zend/Session.php) [698]
Error Code: 7bec12
Stack trace:
#0 /var/www/html/application/libraries/Zend/Session.php(698): session_write_close()
#1 /var/www/html/application/modules/Core/Bootstrap.php(87): Zend_Session::writeClose()
#2 /var/www/html/application/libraries/Engine/Application.php(160): Core_Bootstrap->run()
#3 /var/www/html/application/index.php(194): Engine_Application->run()
#4 /var/www/html/index.php(24): include('/var/www/html/application/index....')
#5 {main}

**Comment by Elshara** [comment]
PHP7 would be awesome.

**Comment by RaymondBenc** [comment]
@hasithaaa thanks for providing the stack trace. We've got this reported internally and we are looking into this.

**Comment by Elshara** [comment]
:)

**Comment by shivani2910** [comment]
This is implemented for 4.9.0 when it is released.

**Comment by Elshara** [comment]
Awesome!
Welcome back.
